agent/src/main/java/com/teamscale/jacoco/agent/options/AgentOptions.java

```diff
@@ -269,9 +262,6 @@ public String getObfuscatedOptionsString() {
 			validateLoggingConfig(validator);
 		}
 
-		validator.isFalse(uploadUrl == null && !additionalMetaDataFiles.isEmpty(),
```

Comment:
See discussion in https://cqse.slack.com/archives/CL0REH3FW/p1705588368099629
The `additionalMetaDataFiles` are passed directly to e.g. the `ArtifactoryUploader` which relies on the URL from `ArtifactoryConfig`, which is a different option than the `upload-url`.
I am not sure about the reason for this check here, since i.m.o. `additionalMetaDataFiles` does not rely only on `uploadUrl` (maybe it did before other upload functionalities where added).

Reply:
Ah yes, thanks for moving the comment :) 

I see Fabis reply which suggests that we can delete the `upload-metadata` as well. Correct me if I am wrong, but this suggests that we can get rid of the whole additionalMetaDataFiles. I would wait for his reply and then get back. :)

Reply:
Okay I think we should keep it.